src/main/groovy/org/jetbrains/intellij/IntelliJPlugin.groovy

```diff
@@ -111,6 +112,14 @@ class IntelliJPlugin implements Plugin<Project> {
         def ivyFile = createIvyRepo(project, extension)
         def version = extension.version
 
+        project.configurations.create("provided")
+
+        project.sourceSets.configure {
+            main.compileClasspath += project.configurations.getByName("provided")
+            test.compileClasspath += project.configurations.getByName("provided")
+            test.runtimeClasspath += project.configurations.getByName("provided")
```

Comment:
If dependency is included in main.runtimeClasspath then plugin is copied in to plugin libs. We expect that in runtime it will be provided by idea. As for the test, I am working on proper way to add that dependency in to plugins-test directory. After that change test.runtimeClasspath will not be required anymore